pkg/ccl/importccl/exportcsv.go

```diff
@@ -31,6 +33,88 @@ import (
 const exportFilePatternPart = "%part%"
 const exportFilePatternDefault = exportFilePatternPart + ".csv"
 
+// csvExporter data structure to augment the compression
+// and csv writer, encapsulating the internals to make
+// exporting oblivious for the consumers
+type csvExporter struct {
+	compressor io.WriteCloser
+	buf        *bytes.Buffer
+	csvWriter  *csv.Writer
+}
+
+// Write
+func (c *csvExporter) Write(record []string) error {
+	return c.csvWriter.Write(record)
+}
+
+// Close
+func (c *csvExporter) Close() error {
+	if c.compressor != nil {
+		return c.compressor.Close()
+	}
+	return nil
+}
+
+// Flush
+func (c *csvExporter) Flush() {
+	c.csvWriter.Flush()
+}
+
+// ResetBuffer
+func (c *csvExporter) ResetBuffer() {
```

Comment:
I think this should also be resetting the compressor, right?

Reply:
indeed

Reply:
After digging a bit into `gzip.Writer.Reset()` it doesn't writes anything into buffer while resetting it, just reset state of the gzip writer to its initial state. Therefore, it seems that either order will be fine, while I do agree that resetting buffer first would be more logically sound.